vars/treebuild.groovy

```diff
@@ -35,20 +35,19 @@ def upstreamBuild(def projectCollection, String currentProject, String settingsX
  * @param project a string following the pattern PROJECT_GROUP/PROJECT_NAME, for example kiegroup/drools
  * @param settingsXmlId maven settings xml file id
  * @param goals maven goals
- * @param skipTests boolean to skip tests or not
+ * @param skipTests Boolean to skip tests or not
  */
-def buildProject(String project, String settingsXmlId, String goals, boolean skipTests, String defaultGroup = "kiegroup") {
-    def projectGroupName = getProjectGroupName(project, defaultGroup)
+def buildProject(String project, String settingsXmlId, String goals, Boolean skipTests = null) {
+    def projectGroupName = getProjectGroupName(project)
     def group = projectGroupName[0]
     def name = projectGroupName[1]
 
     println "Building ${group}/${name}"
     sh "mkdir -p ${group}_${name}"
     dir("${env.WORKSPACE}/${group}_${name}") {
         checkoutProject(name, group)
-        maven.runMavenWithSettings(settingsXmlId, goals, skipTests)
+        maven.runMavenWithSettings(settingsXmlId, goals, skipTests != null ? skipTests : new Properties())
```

Comment:
is this correct? if that skipTests is null there will be empty Properties instead of a false?

Reply:
just because the `runMavenWithSettings` method is overridden  https://github.com/Ginxo/jenkins-pipeline-shared-libraries/blob/f49ed111275fc503ae3cf03d0027efcf098a3426/vars/maven.groovy#L3 we want to call the first one

Reply:
ok